src/main/java/net/obvj/performetrics/Performetrics.java

```diff
@@ -186,7 +186,13 @@ public static MonitoredRunnable monitorOperation(Runnable runnable)
     public static MonitoredRunnable monitorOperation(Runnable runnable, Type... types)
     {
         MonitoredRunnable monitoredRunnable = new MonitoredRunnable(runnable, types);
-        monitoredRunnable.run();
-        return monitoredRunnable;
+        try
+        {
+            monitoredRunnable.run();
+        }
+        finally
+        {
+            return monitoredRunnable;
+        }
```

Comment:
This solves the problem but, as a side effect, it hides the exception and lets the code continue, which may be dangerous.

I am afraid we have no escape here but to leave the method as originally designed, document this restriction, and let the user know that using the `MonitoredRunnable` class (instead of this facade) is a better choice when the metric is required even in the event of an exception thrown by the target `Runnable`.

Alternatively, we could evaluate the possibility of a **new** method `monitorQuietly(...)` that users could choose, knowing that it hides an eventual exception (so the adverb "quietly" in the name). But then we would need some new method that could be used to get the exception, at the end.

Reply:
Sure, I think it is better to have a javadoc for that.